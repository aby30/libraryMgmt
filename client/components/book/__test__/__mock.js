export const BookDetails =
{
    "bookId": 1,
    "availableQty": 1,
    "bookName": "BookName 1",
    "imageUrl": "https://res.cloudinary.com/abyy30/image/upload/v1608652378/bookImages/star.jpg",
    "isSelected": 0
};