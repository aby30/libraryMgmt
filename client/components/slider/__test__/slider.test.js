import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Slider from '../index';
import { SliderData } from './__mock.js';

afterEach(cleanup);

describe('Book card', () => {
    const removeFromSlider = jest.fn();
    const borrowBook = jest.fn();
    const hideSlider = jest.fn();

    it('renders with main cta visible', () => {
        const { asFragment } = render(
            <Slider
                sliderTitleText='Slider text'
                sliderItems={SliderData}
                removeFromSlider={removeFromSlider}
                borrowBook={borrowBook}
                hideSlider={hideSlider}
                showCustomMsg='This is custom msg'
            />
        );
        expect(asFragment()).toMatchSnapshot();
    });

    it('renders with main cta hidden', () => {
        const { asFragment } = render(
            <Slider
                sliderTitleText='Slider text'
                sliderItems={SliderData}
                removeFromSlider={removeFromSlider}
                borrowBook={borrowBook}
                hideSlider={hideSlider}
                showCustomMsg='This is custom msg'
                hideMainCta
            />
        );
        expect(asFragment()).toMatchSnapshot();
    })

})