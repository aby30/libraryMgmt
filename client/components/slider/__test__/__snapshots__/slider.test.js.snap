// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Book card renders with main cta hidden 1`] = `
<DocumentFragment>
  <div
    class="slider"
  >
    <div
      class="slider__title"
    >
      Slider text
      <img
        src="https://res.cloudinary.com/abyy30/image/upload/v1608744290/cross_vho0yd.svg"
      />
    </div>
    <div
      class="slider__inner"
    >
      <div />
      <div
        class="slider__limitMsg alert"
      >
        This is custom msg
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Book card renders with main cta visible 1`] = `
<DocumentFragment>
  <div
    class="slider"
  >
    <div
      class="slider__title"
    >
      Slider text
      <img
        src="https://res.cloudinary.com/abyy30/image/upload/v1608744290/cross_vho0yd.svg"
      />
    </div>
    <div
      class="slider__inner"
    >
      <div />
      <div
        class="slider__limitMsg alert"
      >
        This is custom msg
      </div>
    </div>
  </div>
</DocumentFragment>
`;
